pelican/tests/test_contents.py

```diff
@@ -98,6 +98,19 @@ def test_summary_max_length(self):
         page = Page(**page_kwargs)
         self.assertEqual(page.summary, '')
 
+    def test_summary_end_marker(self):
+        # If a :SUMMARY_END_MARKER: is set, and there is no other summary,
+        # generated summary should contain the specified marker at the end.
+        page_kwargs = self._copy_page_kwargs()
+        settings = get_settings()
+        page_kwargs['settings'] = settings
+        del page_kwargs['metadata']['summary']
+        settings['SUMMARY_END_MARKER'] = 'test_marker'
+        settings['SUMMARY_MAX_LENGTH'] = 10
+        page = Page(**page_kwargs)
+        self.assertEqual(page.summary, truncate_html_words(TEST_CONTENT, 10,
+                                                           'test_marker'))
```

Comment:
Additional assert for checking `'test_marker'` is actually in summary would be good:
```python
self.assertIn('test_marker', page.summary)
```